src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -380,44 +400,121 @@ private String setUpPayDateToString(CaseData caseData) {
-                case SUGGESTION_OF_REPAYMENT_PLAN:
-                    return ResponseOneVOneShowTag.ONE_V_ONE_PART_ADMIT_PAY_INSTALMENT;
-                default:
-                    return ResponseOneVOneShowTag.ONE_V_ONE_PART_ADMIT;
-            }
+        if (YES.equals(caseData.getSpecDefenceAdmittedRequired())) {
+            return ResponseOneVOneShowTag.ONE_V_ONE_PART_ADMIT_HAS_PAID;
+        }
+        switch (caseData.getDefenceAdmitPartPaymentTimeRouteRequired()) {
+            case IMMEDIATELY:
+                return ResponseOneVOneShowTag.ONE_V_ONE_PART_ADMIT_PAY_IMMEDIATELY;
+            case BY_SET_DATE:
+                return ResponseOneVOneShowTag.ONE_V_ONE_PART_ADMIT_PAY_BY_SET_DATE;
+            case SUGGESTION_OF_REPAYMENT_PLAN:
+                return ResponseOneVOneShowTag.ONE_V_ONE_PART_ADMIT_PAY_INSTALMENT;
+            default:
+                return ResponseOneVOneShowTag.ONE_V_ONE_PART_ADMIT;
         }
-        return ResponseOneVOneShowTag.ONE_V_ONE_PART_ADMIT_HAS_PAID;
     }
 
     private ResponseOneVOneShowTag setUpOneVOneFlowForFullAdmit(CaseData caseData) {
-        if (caseData.getSpecDefenceFullAdmittedRequired() == NO) {
-            switch (caseData.getDefenceAdmitPartPaymentTimeRouteRequired()) {
-                case IMMEDIATELY:
-                    return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT_PAY_IMMEDIATELY;
-                case BY_SET_DATE:
-                    return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT_PAY_BY_SET_DATE;
-                case SUGGESTION_OF_REPAYMENT_PLAN:
-                    return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT_PAY_INSTALMENT;
-                default:
-                    return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT;
-            }
+        if (YES.equals(caseData.getSpecDefenceFullAdmittedRequired())) {
+            return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT_HAS_PAID;
+        }
+        switch (caseData.getDefenceAdmitPartPaymentTimeRouteRequired()) {
+            case IMMEDIATELY:
+                return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT_PAY_IMMEDIATELY;
+            case BY_SET_DATE:
+                return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT_PAY_BY_SET_DATE;
+            case SUGGESTION_OF_REPAYMENT_PLAN:
+                return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT_PAY_INSTALMENT;
+            default:
+                return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT;
         }
-        return ResponseOneVOneShowTag.ONE_V_ONE_FULL_ADMIT_HAS_PAID;
+    }
+
+    private CallbackResponse validatePaymentDate(CallbackParams callbackParams) {
+
+        CaseData caseData = callbackParams.getCaseData();
+        List<String> errors = new ArrayList<>();
+
+        if (checkPastDateValidation(caseData.getApplicant1RequestedPaymentDateForDefendantSpec())) {
+            errors.add("Enter a date that is today or in the future");
```

Comment:
Maybe use `Collections.singletonList` here (and `Collections.emptyList` in an else if .errors can't handle null?)

Reply:
After discussion, we add 1 to define the capacity of the arraylist